File: Sakila_Net/VB_SakilaLib/DAO/DAOBD/DAOBDFilmCategory.vb
'using System;
Imports  System.Data
'using System.Collections.Generic;
'using System.Configuration;
'using System.Collections;
Imports SakilaLib.Cl.Harrys.Sakila.Model
Imports SakilaLib.Cl.Harrys.BD


Namespace Cl
  Namespace Harrys
    Namespace Sakila
       Namespace DAOBD

          Public class DAOFilmCategory
             Public  Shared Function SelectAll(nPagina as Integer ,cantidad as Integer) as DataTable
                Dim StSql as String = "Select * from filmCategory"
                Return MyBD.getInstance.SqlSelect(StSql, nPagina, cantidad)
        	   End Function

                    Public  Shared Function SelectOne( filmCategory as ClFilmCategory) as Boolean
             Dim StSql as String = "Select * from filmCategory"
            StSql &= " Where  film_id='" & filmCategory.dataFilmId.dataFilmId & "'" _
                           & "   and category_id='" & filmCategory.dataCategoryId.dataCategoryId & "'" _
                        & ""
                        Dim dt as DataTable = MyBD.getInstance.sqlSelect(stSql)
                        If dt.Rows.Count = 0 Then
                            Return False
                        End If
        filmCategory.DataLastUpdate = dt.Rows(0)("last_update").ToString()
                        Return True
                     end Function

        Public  Shared Function SqlInsert( filmCategory as ClFilmCategory)  as Integer
            Dim stSql as String = "insert into FilmCategory (" _
                       & " film_id ,category_id ,last_update" _
                           & " )values (" _
            & "'" & filmCategory.dataFilmId.datafilmId & "'" _
                     & ",'" & filmCategory.dataCategoryId.datacategoryId & "'" _
                     & ",'" & filmCategory.dataLastUpdate & "'" _
             & ")"
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

        Public  Shared Function SqlDelete( filmCategory as ClFilmCategory) as Integer
            
            Dim stSql as String =  "delete from FilmCategory " _
             & " Where  film_id='" & filmCategory.dataFilmId.dataFilmId & "'" _
                           & "   and category_id='" & filmCategory.dataCategoryId.dataCategoryId & "'" _
                       
       
            Return MyBD.getInstance.sqlEjecutar(stSql)
        End Function

       Public   Shared Function SqlUpdate(filmCategory as ClFilmCategory) as Integer
            Dim stSql as String =  "update FilmCategory SET " _
             & " last_update = '" & filmCategory.DataLastUpdate & "'" _
             & " Where  film_id='" & filmCategory.dataFilmId.dataFilmId & "'" _
                           & "   and category_id='" & filmCategory.dataCategoryId.dataCategoryId & "'" _
& ""
            Return MyBD.getInstance.sqlEjecutar(stSql)
            
        end Function
          End Class

         End Namespace  
      End Namespace  
   End Namespace  
End Namespace  

File: Sakila_Net/Sakila_Net/CS_SakilaLib/DAO/DAOBD/DAOBDFilmCategory.cs
using System;
using  System.Data;
using System.Collections.Generic;
using System.Configuration;
using System.Collections;
using Cl.Harrys.Sakila.Model;
using Cl.Harrys.MyBD;


namespace Cl.Harrys.Sakila.DAOBD 
{ 
    public class DAOFilmCategory 
    {
        public  static DataTable selectAll(int nPagina  ,int cantidad ) 
        {
            String StSql = "Select  film_id ,category_id ,last_update  from filmCategory";
            return MyBD.MyBD.getInstance().sqlSelect(StSql, nPagina, cantidad);
        }

        public  static Boolean SelectOne( ClFilmCategory filmCategory)  
        {
            String stSql = "Select  film_id ,category_id ,last_update  from filmCategory";
            stSql += " Where  film_id='" + filmCategory.FilmId.FilmId + "'" 
                           + "   and category_id='" + filmCategory.CategoryId.CategoryId + "'" 
                        + "";
            DataTable dt = MyBD.MyBD.getInstance().sqlSelect(stSql,1,1);
            if (dt.Rows.Count == 0) 
                return false;
                        
        filmCategory.LastUpdate = DateTime.ParseExact(dt.Rows[0]["last_update"].ToString(), "yyyy-MM-dd HH:mm:ss,fff",System.Globalization.CultureInfo.InvariantCulture);
            return true;
        }

        public  static int sqlInsert( ClFilmCategory filmCategory)  
        {
            String stSql = "insert into FilmCategory (" 
                       + " film_id ,category_id ,last_update" 
                           + " )values (" 
            + "'" + filmCategory.FilmId.FilmId + "'" 
                     + ",'" + filmCategory.CategoryId.CategoryId + "'" 
                     + ",'" + filmCategory.LastUpdate + "'" 
             + ")";
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

        public  static int sqlDelete( ClFilmCategory filmCategory) {
            
            String stSql  =  "delete from FilmCategory " 
             + " Where  film_id='" + filmCategory.FilmId.FilmId + "'" 
                           + "   and category_id='" + filmCategory.CategoryId.CategoryId + "'" 
                       ;
       
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
        }

       public   static int sqlUpdate(ClFilmCategory filmCategory ) {
            String stSql  =  "update FilmCategory SET " 
             + " last_update = '" + filmCategory.LastUpdate + "'" 
             + " Where  film_id='" + filmCategory.FilmId.FilmId + "'" 
                           + "   and category_id='" + filmCategory.CategoryId.CategoryId + "'" 
                 + "";
            return MyBD.MyBD.getInstance().sqlEjecutar(stSql);
            
        }
   }

}
